source/PopNGo/PopNGo/PopNGo/wwwroot/js/favorites.js

```diff
@@ -0,0 +1,64 @@
+import { formatStartTime } from './Helper-Functions/formatStartTime.js';
+
+
+function fetchUserFavorites() {
+    fetch('/api/FavoritesApi/GetUserFavorites')
+        .then(response => {
+            if (response.status === 401) {
+                // Unauthorized, tell the user to log in or sign up
+                document.getElementById('login-prompt').style.display = 'block';
+                throw new Error('Unauthorized');
+            }
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            return response.json();
+        })
+        .then(data => {
+            if (data.length === 0) {
+                document.getElementById('no-favorites-message').style.display = 'block';
+                return;
+            }
```

Comment:
If the above is implemented, this can be removed. As-is, the code never gets here to check this.